### PR TITLE
Enhance Disk Expansion Logic and Microshift ISO Generation Guide

### PR DESCRIPTION
**Pull request by ritanorinho**
**Changes**
- Implemented an automatic restart mechanism for the Microshift service upon creation of the lvmd.yaml file and the essential files required for external disk usage.
- Included a guide outlining the necessary steps to generate a Microshift ISO.

### CODE REVIEW

File: mesinabox/microshiftIsoGeneration/howTo.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        - `ssh-copy-id <user>@<image builder IP> `
+    - Inject the pull secret as an Ansible variable so [get the pull secret from the Red Hat Console](https://console.redhat.com/openshift/install/pull-secret).
+        - `ansible-vault create vars/secrets.yml`
+        - Then add the variable microshift_pull_secret in that file with the right value and then save it.
+            - `microshift_pull_secret: '<YOUR PULL SECRET>'`
+    - Create CMOS App.
+        - Clone install-scripts repository into Rhel VM
+            - `git clone https://github.com/criticalmanufacturing/install-scripts.git`
+        - Copy files from mesinabox/stack in this repo to the following directory edge-demos/demos/microshift-disconnected/files/manifests/cmos
+        - Based on the content present in [this file](https://github.com/traefik/traefik/blob/v2.10/docs/content/reference/dynamic-configuration/kubernetes-crd-definition-v1.yml), create a yaml file for each resource in the cmos folder.
```
[the remark this answers is not in the document]
Comment (ritanorinho) [reply mesinabox/microshiftIsoGeneration/howTo.md:41]: yes, based on the content presented in this file

File: mesinabox/kioskapp/server.js [hunk 2]
```diff
@@ -363,32 +362,67 @@ app.post('/expandDisk', (req, res) => {
...
+            res.status(500).send("There was an issue while trying to expand the disk!");
+          }})
+      } else {
+       await writeToFifo(pipePath, createPVScript);
+       readFifoOutput(res, async (output) => {
+          if (output === '0') {
+            res.status(200).send("System is restarting. Refresh it in 1 minute.");
+            await writeToFifo(pipePath, restartMicroshiftService);
+          } else {
+            res.status(500).send("There was an issue while trying to expand the disk! 2");
```
[the remark this answers is not in the document]
Comment (ritanorinho) [reply mesinabox/kioskapp/server.js:396]: I need to remove the "2" but this message was previously aligned.

File: mesinabox/kioskapp/server.js [hunk 3]
```diff
@@ -363,32 +362,67 @@ app.post('/expandDisk', (req, res) => {
         if (fs.existsSync(outputPath)) {
           fs.unlinkSync(outputPath); //delete the output file
         }
-        if (data === '0') {
-          res.status(200).send("Disk successfully expanded!");
-        } else {
-          res.status(500).send("There was an issue while trying to expand the disk!");
-        }
+        console.log("data ", data);
```
[the remark this answers is not in the document]
Comment (ritanorinho) [reply mesinabox/kioskapp/server.js:365]: yep

File: mesinabox/kioskapp/server.js [hunk 4]
```diff
@@ -363,32 +362,67 @@ app.post('/expandDisk', (req, res) => {
...
+      if (output === '0') {
+        await writeToFifo(pipePath, expandScript);
+        readFifoOutput(res, async(output) => {
+          if (output === '0') {
+            res.status(200).send("Disk successfully expanded!");
+          } else {
+            res.status(500).send("There was an issue while trying to expand the disk!");
+          }})
+      } else {
+       await writeToFifo(pipePath, createPVScript);
```
[the remark this answers is not in the document]
Comment (ritanorinho) [reply mesinabox/kioskapp/server.js:390]: Nope, the system will always restart.

File: mesinabox/microshiftIsoGeneration/howTo.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
+# How to generate Microshift ISO
+
+1. Get [RHEL 9 Binary DVD .iso](https://access.redhat.com/downloads/content/rhel).
```
[the remark this answers is not in the document]
Comment (ritanorinho) [reply mesinabox/microshiftIsoGeneration/howTo.md]: done

**Review by afjsilva — APPROVED** [review]

**Pull request merged**